libraries/chain/include/eosio/chain/transaction_metadata.hpp

```diff
@@ -73,6 +69,10 @@ class transaction_metadata {
       fc::microseconds signature_cpu_usage()const { return _sig_cpu_usage; }
       const flat_set<public_key_type>& recovered_keys()const { return _recovered_pub_keys; }
       size_t get_estimated_size() const;
+      const trx_type get_trx_type() const { return _trx_type; };
+      const bool implicit() const { return _trx_type == trx_type::implicit; };
+      const bool scheduled() const { return _trx_type == trx_type::scheduled; };
+      const bool is_dry_run() const { return _trx_type == trx_type::dry_run; };
```

Comment:
Same question, why the `const` on return types?

Reply:
Will remove `const`.